src/test/regress/expected/multi_hash_pruning.out

```diff
@@ -188,19 +188,86 @@ DEBUG:  Plan is router executable
      0
 (1 row)
 
--- Check that we don't support pruning for ANY (array expression) and give
--- a notice message when used with the partition column
-SELECT count(*) FROM orders_hash_partitioned
-	WHERE o_orderkey = ANY ('{1,2,3}');
-NOTICE:  cannot use shard pruning with ANY/ALL (array expression)
-HINT:  Consider rewriting the expression with OR/AND clauses.
-NOTICE:  cannot use shard pruning with ANY/ALL (array expression)
-HINT:  Consider rewriting the expression with OR/AND clauses.
+SET client_min_messages TO DEFAULT;
+-- Check that we support running for ANY/IN
+CREATE TABLE sample_subquery_table(id int);
+CREATE TABLE sample_main_table(id int);
+SELECT create_distributed_table('sample_subquery_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('sample_main_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+INSERT INTO sample_subquery_table VALUES(1);
+INSERT INTO sample_subquery_table VALUES(5);
+INSERT INTO sample_main_table VALUES(1);
+INSERT INTO sample_main_table VALUES(5);
+-- Check that we support runing for ANY/IN with literal.
+SELECT count(*) FROM sample_main_table
+	WHERE id = ANY ('{1,2,3}');
+ count 
+-------
+     1
+(1 row)
+
+SELECT count(*) FROM sample_main_table
+	WHERE id IN (1,2,3);
+ count 
+-------
+     1
+(1 row)
+
+-- Check whether we can deal with null arrays
+SELECT count(*) FROM sample_main_table
+	WHERE id IN (NULL);
  count 
 -------
      0
 (1 row)
 
+SELECT count(*) FROM sample_main_table
+	WHERE id = ANY (NULL);
+ count 
+-------
+     0
+(1 row)
+
+SELECT count(*) FROM sample_main_table
+	WHERE id IN (NULL) OR TRUE;
+ count 
+-------
+     2
+(1 row)
+
+SELECT count(*) FROM sample_main_table
+	WHERE id = ANY(NULL) OR TRUE;	
+ count 
+-------
+     2
+(1 row)
+
+-- Check whether we support IN/ANY in subquery
+SELECT count(*) FROM sample_main_table WHERE ID IN (SELECT id FROM sample_subquery_table);
+ count 
+-------
+     2
+(1 row)
+
+SELECT count(*) FROM sample_main_table WHERE ID IN (SELECT id FROM sample_subquery_table);
+ count 
+-------
+     2
+(1 row)
+
+DROP TABLE sample_subquery_table;
+DROP TABLE sample_main_table;
+SET client_min_messages TO DEBUG2;
 -- Check that we don't show the message if the operator is not
```

Comment:
Couple points:
- do we really need new tables for these
- I don't think these cover append & range partitioned tables/pruning